src/AppChooser/Dialog.vala

```diff
@@ -83,89 +83,97 @@ public class AppChooser.Dialog : Hdy.Window {
-            "application-default-icon"
-        );
-        placeholder.show_all ();
+        var placeholder = new Granite.Placeholder (_("No installed apps can open %s").printf (content_description)) {
+            description = _("New apps can be installed from AppCenter"),
+            icon = new ThemedIcon ("application-default-icon")
+        };
 
         listbox = new Gtk.ListBox () {
-            expand = true
+            hexpand = true,
+            vexpand = true
         };
+        listbox.add_css_class (Granite.STYLE_CLASS_RICH_LIST);
         listbox.set_placeholder (placeholder);
 
-        var scrolled_window = new Gtk.ScrolledWindow (null, null);
-        scrolled_window.add (listbox);
+        var scrolled_window = new Gtk.ScrolledWindow () {
+            child = listbox
+        };
 
-        var frame = new Gtk.Frame (null);
-        frame.add (scrolled_window);
+        var frame = new Gtk.Frame (null) {
+            child = scrolled_window
+        };
 
-        var cancel = new Gtk.Button.with_label (_("Cancel"));
+        cancel_button = new Gtk.Button.with_label (_("Cancel"));
 
         open_button = new Gtk.Button.with_label (_("Open")) {
-            can_default = true
+            receives_default = true
         };
-        open_button.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
+        open_button.add_css_class (Granite.STYLE_CLASS_SUGGESTED_ACTION);
 
-        var button_box = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL) {
-            layout_style = Gtk.ButtonBoxStyle.END,
-            margin_top = 12,
-            spacing = 6
+        var button_box = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0) {
+            halign = Gtk.Align.END
         };
-
-        button_box.add (cancel);
-        button_box.add (open_button);
+        button_box.append (cancel_button);
+        button_box.append (open_button);
+        button_box.add_css_class ("dialog-action-area");
 
         var grid = new Gtk.Grid () {
             orientation = Gtk.Orientation.VERTICAL,
             column_spacing = 12,
-            row_spacing = 6,
-            margin = 12
+            row_spacing = 6
         };
 
         grid.attach (overlay, 0, 0, 1, 2);
         grid.attach (primary_label, 1, 0);
         grid.attach (secondary_label, 1, 1);
         grid.attach (frame, 0, 3, 2);
-        grid.attach (button_box, 1, 4);
+        grid.add_css_class (Granite.STYLE_CLASS_DIALOG_CONTENT_AREA);
 
-        var window_handle = new Hdy.WindowHandle ();
-        window_handle.add (grid);
+        var box = new Gtk.Box (Gtk.Orientation.VERTICAL, 0);
+        box.append (grid);
+        box.append (button_box);
+        box.add_css_class ("dialog-vbox");
 
-        add (window_handle);
-        type_hint = Gdk.WindowTypeHint.DIALOG;
+        var window_handle = new Gtk.WindowHandle () {
+            child = box
+        };
+
+        child = window_handle;
+
+        modal = true;
+        decorated = false;
```

Comment:
This is wrong, we need decorated to have round conners.

Reply:
Ah thank you! fixed :)